catalyst-cooperative/pudl
# Merge dev into main for 2023-08-11

- Fix small typos
- Update black requirement from <23.4,>=22.0 to >=22.0,<23.8
- Integrate Form 60 into dbf extraction (2006-2020)
- Fix XBRL Ferc Form 2 year parameters
- Update release notes
- Add old FERC 60 DBF data to the Datasette deployment.
- Add EIA860 2022 data
- Add release notes for eia860 2022 data
- Make generator_id fill in hack fail if the value is filled in later.
- Preliminary integration of EIA923 2022 data
- Update dagster requirement from <1.4,>=1.2.2 to >=1.2.2,<1.5
- Update dask requirement from <2023.7.1,>=2021.8 to >=2021.8,<2023.7.2
- Update recordlinkage requirement from <0.16,>=0.14 to >=0.14,<0.17
- Update dagster and dagster-webserver (formerly dagit) together.
- Replace references to dagit with dagster UI and dagster-webserver
- Rename dagster UI images
- Add FERC60 to data access docs
- Fixes to the glue_test and ferc1_eia functions to accomodate new data quirks, mapping for new EIA plants and utils
- Add release notes
- Remove unecessary comment
- Add in EIA860m data and make EIA fast tests include 2020 so as not to break the ferc1_eia fast outputs. There is only training data for FERC-EIA up through 2020 so that year must be included in both tests.
- Replace deprecated pkg_resources usage with importlib.metadata.
- Update to using modern importlib.resources API.
- Update minmax row count after inclusion of eia860m
- Add unmapped utilities to csv
- Add unmapped plants to the csv
- Remove redundant code
- Remove 2021 from the fast etl so there are only two years and hopefully the ci tests don't get overwhelmed
- Remove outdated explanation about our python packaging.
- Update jedi requirement from <0.19,>=0.18 to >=0.18,<0.20
- Update flake8 requirement from <6.1,>=4.0 to >=4.0,<6.2
- Update sphinx requirement from !=5.1.0,<7.1,>=4 to >=4,!=5.1.0,<7.2
- Add some ferc1_eia training data for 2021 and make EIA fast etl training data consecutive
- [pre-commit.ci] pre-commit autoupdate
- Update jupyter resource monitoring package
- Ignore redefition of function to be returned as Callable.
- Ignore redefition of function to be returned as Callable.
- Remove 2021 training data and revert to EIA fast etl data from 2020 and 2022
- Replace == with is in dictionary comprehension in glue tests
- Add 2021 to EIA fast etl to test memory fail on ci
- Remove 2021 from EIA fast etl and x fail ferc1_eia test to see if the ci passes. This would mean that the ferc1_eia test is the source of the memory error
- Add back 2021 to EIA fast etl to try and get the tests to run
- Make EIA fast etl 2020 and 2021 to see if that fixes memory issue in the ci
- Comment out 2022 spot fix in EIA that was causing the fail. Try again to see if ci passes with just 2020 and 2021
- Revert back to 2020 and 2022 data, try commenting out memory intensive test_original_gf_vs_the_allocated_by_gens_gf to see if that works
- Small fixes to remove blank lines from training data and add caveat to 2022 spot fix in eia860 transform so that it won't fail if 2022 isn't in the fast etl
- Update grpcio requirement from <1.55.0 to <1.56.0
- Update dask requirement from <2023.7.2,>=2021.8 to >=2021.8,<2023.8.1
- Instead of commenting out test_original_gf_vs_the_allocated_by_gens_gf function, add a conditional that will not run it as part of the ci
- Add comment to conditional blocking test_original_gf_vs_the_allocated_by_gens_gf so we know what's going on
- [pre-commit.ci] pre-commit autoupdate
- Update minmax row lengths for mcoe tables
- Update one minmax rows and increase the upper bound for gas prices due to 2022 spikes
- Remove inf values from fuel_cost_per_mwh col in mcoe
